# Import sth

In [1]:
import math
import matplotlib.pyplot as plt

# Boundary condition

## Air inlet velocity for 

In [2]:
# to compute the inlet velocity of air
diammeter_fuel_inner = 4*1e-3
diammeter_fuel_outer = 24*1e-3
diammeter_air_inner = 70*1e-3
airflow_section_area = 3.14*(diammeter_air_inner**2 - diammeter_fuel_outer**2)/4

airflow_volume_speed = 0.000747329
 #0.00597863

airflow_velocity_speed = airflow_volume_speed/airflow_section_area
print(f'the inlet velocity of air is :{airflow_velocity_speed} m/s')

the inlet velocity of air is :0.2201691639611824 m/s


## Fuel inlet velocity for fluent input

In [136]:
# to compute the fuel inlet velocity of air

fuelflow_section_area = 3.14*diammeter_fuel_inner**2 /4

fuelflow_volume_speed = 0.0002512

fuelflow_velocity_speed = fuelflow_volume_speed/fuelflow_section_area
print(f'the velocity of fuel is :{fuelflow_velocity_speed} m/s')

the velocity of fuel is :20.0 m/s


# Swril number (Sn)

|Swirl type| Diameter (mm) | Vane angel(°) |Vane number|Swirl number|
|---|-----|-------|-----------|-----------|
| Type 1 | Inner: 13, outer 34.4|45 | 12 | 0.736 |
| Type 2 | -|60 | 12 | 1.27 |

In [139]:
# refer paper: {Hayakawa, 2017 #1003} Formula (1), Table 1 - Swirler specifications

# input the diameter and angle of the swirl vane
vane_angle = 60 #in angle
#angle_in_radian = vane_angle/360*(2*math.pi) # angle in radian
vane_angle_in_radian = math.radians(vane_angle) # angle in radian
number_of_vane = 6 # 叶片数（并不参与计算）
diameter_of_inner = 13 # d_h, the hub diameter in mm
diameter_of_outer = 34.4 # d_t, the swirl tip diameter in mm
d_ratio = diameter_of_inner/diameter_of_outer

# formula
swirl_number = (2/3)*((1-d_ratio**3)/(1-d_ratio**2))*math.tan(vane_angle_in_radian)
swirl_number_round = round(swirl_number,3) # Round a number to only three decimals

# output
print(f"彼时\n旋流器内、外直径大小为：{diameter_of_inner}、{diameter_of_outer} （mm）；\n角度及弧度大小为：{vane_angle} (°)、{vane_angle_in_radian} (rad)\n旋流数为(Sn): {swirl_number_round}")

彼时
旋流器内、外直径大小为：13、34.4 （mm）；
角度及弧度大小为：60 (°)、1.0471975511965976 (rad)
旋流数为(Sn): 1.274


In [135]:
# 角度转换弧度
a = 45 # 角度
in_radians = 45/360*2*3.14
print(in_radians)
print(math.radians(45))

0.785
0.7853981633974483


甲烷掺氢旋流钝体之——旋流结构参数
|Swirl type| Diameter (mm) | Vane angel(°) |Vane number|Swirl number|
|---|-----|-------|-----------|-----------|
| Swirl 1 | inner: 25, outer: 70 |30 | 6 | xxx |
| Swirl 2 | - | 45 | 6 | xxx |
| Swirl 3 | - | 60 | 6 | xxx |

In [147]:
# 暂定: 旋流器结构高度为20mm

# input the diameter and angle of the swirl vane
vane_angle = 75 #in angle
#angle_in_radian = vane_angle/360*(2*math.pi) # angle in radian
vane_angle_in_radian = math.radians(vane_angle) # angle in radian
number_of_vane = 6 # 叶片数（并不参与计算）
diameter_of_inner = 25 # d_h, the hub diameter in mm
diameter_of_outer = 70 # d_t, the swirl tip diameter in mm
d_ratio = diameter_of_inner/diameter_of_outer

# formula
swirl_number = (2/3)*((1-d_ratio**3)/(1-d_ratio**2))*math.tan(vane_angle_in_radian)
swirl_number_round = round(swirl_number,3) # Round a number to only three decimals

# output
print(f"彼时\n旋流器内、外直径大小为：{diameter_of_inner}、{diameter_of_outer} （mm）；\n角度及弧度大小为：{vane_angle} (°)、{vane_angle_in_radian} (rad)\n计算旋流数大小为(Sn): {swirl_number_round}")

彼时
旋流器内、外直径大小为：25、70 （mm）；
角度及弧度大小为：75 (°)、1.3089969389957472 (rad)
计算旋流数大小为(Sn): 2.722


|Swirl type| Diameter (mm) | Vane angel(°) |Vane number|Swirl number|
|---|-----|-------|-----------|-----------|
| Swirl 1 | inner: 25, outer: 70 |30 | 6 | 0.421 |
| Swirl 2 | - | 45 | 6 | 0.729 |
| Swirl 3 | - | 60 | 6 | 1.263 |
| Swirl 4 | - | 75 | 6 | 2.722 |

甲烷掺氢旋流钝体之——钝体结构参数（待完善-2023.04.19）
| Bluff body(BB) type | Height (mm) | BB angel(°) | Vane number | Swirl number |
|---|-----|-------|-----------|-----------|
| BB 1 | inner: 25, outer: 70 |30 | 6 | xxx |
| BB 2 | - | 45 | 6 | xxx |
| BB 3 | - | 60 | 6 | xxx |

## Re计算
